### PR TITLE
[WNMGDS-1817, WNMGDS-1818, WNMGDS-1861] remove old doc site content, update a11y tests

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

- Removed old doc site content
- Removed old doc site yarn tasks
- Removed old doc site build functionality from `design-system-scripts`
- Updated `design-system-scripts` and our a11y tests extensively, now uses `storybook-static` builds to test against
- Changed `build:gatsby` to `build:docs`
- Changed `serve:gatsby` to `serve:docs`
- Changed `start` to start gatsby in dev mode
- Removed existing a11y tests, special functionality can be worked in again, in the future

## How to test

Make sure all our build and testing tasks work!

`yarn install && yarn build && yarn build:healthcare && yarn build:medicare`
`yarn build-storybook:gatsby && yarn build:gatsby && yarn serve:gatsby`
`yarn start`
`yarn storybook`
`yarn test:unit`
`yarn test:a11y core` -- note there are violations but all tests show as passing, this is to be fixed in a future ticket
`yarn test:a11y (healthcare/medicare)`

### CODE REVIEW

File: packages/design-system-scripts/gulp/common/constants.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const path = require('path');
 
 const CORE_SOURCE_PACKAGE = '@cmsgov/design-system';
-const CORE_DOCS_PACKAGE = '@cmsgov/design-system-docs';
+const CORE_DOCS_PACKAGE = '@cmsgov/cms-design-system-docs';
```
Comment (zarahzachz) [inline packages/design-system-scripts/gulp/common/constants.js:4]: Curious what this docs package is? Thought we were deleting the design-system-docs and this looks like a new/different docs package

**Review by zarahzachz — COMMENTED** [review]
Looks good! Left a few documentation-type questions for feedback.

**Comment by zarahzachz** [comment]
Noticed the test steps have some outdated scripts. Might need to update for others to test more easily.

File: packages/design-system-scripts/gulp/common/constants.js (hunk 1 above)
Comment (forrestbaer) [reply packages/design-system-scripts/gulp/common/constants.js:4]: It's just the current name of the ./docs in it's package.json, this line can likely be removed, i'll take a look.

File: package.json [hunk 2]
```diff
@@ -18,17 +18,14 @@
...
-    "build-docs": "yarn cmsds build-docs",
-    "build-docs:healthcare": "yarn cmsds build-docs --config cmsds.healthcare.config.js",
-    "build-docs:medicare": "yarn cmsds build-docs --config cmsds.medicare.config.js",
-    "build:gatsby": "yarn --cwd ./packages/docs build",
-    "serve:gatsby": "yarn --cwd ./packages/docs serve",
-    "start:gatsby": "yarn --cwd ./packages/docs develop",
-    "test": "yarn test:unit && yarn test:a11y",
+    "build:docs": "yarn --cwd ./packages/docs build",
+    "serve:docs": "yarn --cwd ./packages/docs serve",
+    "start": "yarn --cwd ./packages/docs develop",
```
[the remark this answers is not in the document]
Comment (forrestbaer) [reply package.json:23]: Yeah I actually am going to followup this PR with a changing of some of our npm scripts to all use the same `verb:noun` format, which will mean updating some of the jenkins scripts but I wanted to put it all in another PR. Good callout.

File: README.md [hunk 3]
```diff
@@ -16,12 +16,11 @@
...
 | [CMS Design System](packages/design-system)                | The core CSS, JS, and React components for the design system. <br> [![@cmsgov/design-system](https://img.shields.io/npm/v/@cmsgov/design-system.svg?label=@cmsgov%2Fdesign-system)](https://www.npmjs.com/package/@cmsgov/design-system)                                                                                                                                                                                                                     |
-| [Design System Documentation](packages/design-system-docs) | Markdown files containing documentation for the core design system site. These files are used by `@cmsgov/design-system-scripts` to generate our documentation site. <br> [![@cmsgov/design-system](https://img.shields.io/npm/v/@cmsgov/design-system.svg?label=@cmsgov%2Fdesign-system-docs)](https://www.npmjs.com/package/@cmsgov/design-system-docs)                                                                                                    |
```
[the remark this answers is not in the document]
Comment (forrestbaer) [reply README.md:24]: Yeah good idea, added it.

**Comment by forrestbaer** [comment]
> Noticed the test steps have some outdated scripts. Might need to update for others to test more easily.

Where exactly? I'm not sure what to look at.

File: packages/design-system-scripts/gulp/sass.js [hunk 4]
```diff
@@ -55,16 +55,6 @@ function compileSass(src, dest, includePaths, browserSync) {
     )
     .pipe(gulp.dest(dest));
 
-  if (browserSync) {
```
Comment (pwolfert) [inline packages/design-system-scripts/gulp/sass.js:58]: Nice! I think that means we can also remove it from the `compileSass` function and also remove the `browser-sync` dependency. It looks like this wasn't even hooked up before 😕

File: packages/design-system-scripts/gulp/build.js [hunk 5]
```diff
@@ -212,11 +206,11 @@ function compileJs(dir, options, changedPath) {
 async function bundleJs(dir) {
   logTask('🚜 ', 'Running Webpack statically');
   try {
-    const config = await createCdnWebpackConfig(dir);
```
Comment (pwolfert) [inline packages/design-system-scripts/gulp/build.js:215]: Nice catch

File: packages/design-system-scripts/gulp/sass.js (hunk 4 above)
Comment (forrestbaer) [reply packages/design-system-scripts/gulp/sass.js:58]: Ahh nice, removed! That took a bunch of junk with it, nice.

File: packages/design-system-scripts/jest/a11y.global-setup.js [hunk 6]
```diff
@@ -1,27 +1,15 @@
...
-  log(chalk.green('done ✓'));
-}
-
 async function startServer() {
-  log(chalk.green('\nStarting local server hosting production build...'));
+  console.log('\n');
+  log(chalk.green('Starting local web server in ./storybook-static/ ...'));
   const server = new StaticServer({
-    rootPath: process.env.BUILD_PATH,
+    rootPath: './storybook-static/',
```
Comment (pwolfert) [inline packages/design-system-scripts/jest/a11y.global-setup.js:12]: I like this setup 👍. Now that I know that the config file gets loaded before this one executes, this makes a lot of sense.

File: packages/docs/package.json [hunk 7]
```diff
@@ -33,7 +33,8 @@
     "prettier": "^2.6.2",
     "prismjs": "^1.11.0",
     "react-helmet": "^6.1.0",
-    "sass": "^1.43.4"
+    "sass": "^1.43.4",
+    "url-join": "^4.0.1"
```
Comment (pwolfert) [inline packages/docs/package.json:37]: Good catch

**Review by pwolfert — APPROVED** [review]
These changes work great! I don't love the idea of disabling these assertions on the main branch. We should consider merging this into a feature branch and fixing the violations soon.

**Comment by forrestbaer** [comment]
> These changes work great! I don't love the idea of disabling these assertions on the main branch. We should consider merging this into a feature branch and fixing the violations soon.

**Comment by forrestbaer** [comment]
Oh no, didn't mean to close..

**Pull request merged**